ui/src/variables/components/VariableFormContext.tsx

```diff
@@ -0,0 +1,111 @@
+// Libraries
+import React, {PureComponent} from 'react'
+import {connect} from 'react-redux'
+
+// Utils
+import {
+  extractVariablesList,
+  extractVariableEditorName,
+  extractVariableEditorType,
+  extractVariableEditorQuery,
+  extractVariableEditorMap,
+  extractVariableEditorConstant,
+} from 'src/variables/selectors'
+
+// Actions
+import {
+  updateName,
+  updateType,
+  updateQuery,
+  updateMap,
+  updateConstant,
+  clearEditor,
+  createVariable,
+} from 'src/variables/actions'
+
+// Component
+import VariableForm from 'src/variables/components/VariableForm'
+
+// Types
+import {IVariable as Variable} from '@influxdata/influx'
+import {AppState, VariableArguments, VariableArgumentType} from 'src/types'
+
+interface ComponentProps {
+  onHideOverlay?: () => void
+  initialScript?: string
+}
+
+interface DispatchProps {
+  onCreateVariable: (
+    variable: Pick<Variable, 'name' | 'arguments' | 'selected'>
+  ) => void
+  onNameUpdate: typeof updateName
+  onTypeUpdate: typeof updateType
+  onQueryUpdate: typeof updateQuery
+  onMapUpdate: typeof updateMap
+  onConstantUpdate: typeof updateConstant
+  onEditorClose: typeof clearEditor
+}
+
+interface StateProps {
+  variables: Variable[]
+  name: string
+  variableType: VariableArgumentType
+  query: VariableArguments
+  map: VariableArguments
+  constant: VariableArguments
+}
+
+type Props = ComponentProps & DispatchProps & StateProps
+
+class VariableFormContext extends PureComponent<Props> {
+  render() {
+    const props = {
+      onHideOverlay: this.handleHideOverlay,
+      ...this.props,
+    }
+    return <VariableForm {...props} />
+  }
+
+  private handleHideOverlay() {
+    const {onHideOverlay, onEditorClose} = this.props
+
+    onEditorClose()
+    onHideOverlay()
+  }
+}
+
+const mstp = (state: AppState): StateProps => {
+  const variables = extractVariablesList(state),
```

Comment:
Love all the selectors.

Silly nit but is the prefix `extract` adding value to these function names?

Reply:
not at all! i was trying to follow the convention, but extractions/segmentation/whatever is a function of their include path, so it's a bit redundant.